scipy/optimize/_cobyqa_py.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+
+from ._optimize import _check_unknown_options
+
+__all__ = []
+
+
+def _minimize_cobyqa(fun, x0, args=(), bounds=None, constraints=(),
+                     callback=None, disp=False, maxfev=1000, maxiter=1000,
```

Comment:
I do not know COBYQA in detail but in my experience for higher dimensional problems, derivative-free solvers need often a much higher function evaluation budget. What about setting it automatically to something like 1000 * dimensionality of the problem?

Reply:
Also could you add tests that with a very tiny budget for `maxiter` and `maxfev` the results object returns `success=False` and that the actual number of `maxfev` stays below the provided one then still? I saw that your tests contain logic to count the number of function evaluations, thanks for that!

Reply:
> I do not know COBYQA in detail but in my experience for higher dimensional problems, derivative-free solvers need often a much higher function evaluation budget. What about setting it automatically to something like 1000 * dimensionality of the problem?

You are right. In the COBYQA package, I use `500 * n` for the default `maxfev`, and `1000 * n` for the default `maxiter`. I agree these values are much better, I just copied these default from COBYLA interface. I will update these.

Reply:
> Also could you add tests that with a very tiny budget for `maxiter` and `maxfev` the results object returns `success=False` and that the actual number of `maxfev` stays below the provided one then still? I saw that your tests contain logic to count the number of function evaluations, thanks for that!

Great idea! I will do that also.

Reply:
This should be done in e737b9c1518aed996a4798b94ce1bc67692cc780. I let you resolve this conversation if you are happy with the modifications.

Reply:
I am happy with the changes but do not have "resolve" permissions.